### PR TITLE
V3: fix laggy filter search

### PR DESCRIPTION
**Pull request by tuanddd**
## What this PR does
- [x] Fix laggy filter search by virtualizing the list when there are too many items

### CODE REVIEW

**Comment by sampotts** [comment]
I'm seeing an issue where the item heights shrink when the list is expanded. Was it ok when you tested?

https://user-images.githubusercontent.com/719092/103604274-ac3dfe80-4f64-11eb-940b-b2aed0d648f7.mp4

**Comment by tuanddd** [comment]
> I'm seeing an issue where the item heights shrink when the list is expanded. Was it ok when you tested?
> 
>  CleanShot.2021-01-05.at.14.43.52.mp4

Let me take a look.

File: packages/search-ui/src/Filter/ListFilter.tsx [hunk 1]
```diff
@@ -79,27 +77,60 @@ const ListFilter = (props: Omit<ListFilterProps, 'type'>) => {
     return list;
   }, [JSON.stringify(filtered), JSON.stringify(selected), pinSelected, sort, sortAscending]);
 
-  const items = slice && !expanded ? sortedItems.slice(0, limit) : sortedItems;
+  const rowVirtualizer = useVirtual({
+    parentRef,
+    size: sortedItems.length,
+    // Recalculate item's height on change
+    estimateSize: React.useCallback(() => 21, [JSON.stringify(selected)]),
```
Comment (sampotts) [inline packages/search-ui/src/Filter/ListFilter.tsx:84]: How can we test this is working?

File: packages/search-ui/src/Filter/ListFilter.tsx [hunk 2]
```diff
@@ -114,22 +145,27 @@ const ListFilter = (props: Omit<ListFilterProps, 'type'>) => {
...
+            onChange={filterSearchOnChange}
             disableDefaultStyles={disableDefaultStyles}
           />
         </CoreBox>
       ) : null}
 
-      <CoreBox id={`list-${name}`} className={customClassNames.filter?.list?.container}>
+      <CoreBox
+        id={`list-${name}`}
+        css={slice && expanded ? tw`overflow-auto h-96` : tw``}
```
Comment (sampotts) [inline packages/search-ui/src/Filter/ListFilter.tsx:156]: Shouldn't this be `max-h-96`? Otherwise, we're fixing the height when it may not be required?

**Review by sampotts — APPROVED** [review]

**Pull request merged**